### ISSUE TITLE
CMake targets don't seem to be working

### ISSUE THREAD

**Issue by thatbakamono** [chalet-org/chalet#160]
I specified the target as follows:
```json
		"glfw": {
			"kind": "cmakeProject",
			"location": "${externalDir}/glfw",
			"targets": ["glfw"]
		},
```
but it didn't really change anything as tests and examples are still being built:
```
   [1/106] Building C object src\CMakeFiles\glfw.dir\context.c.obj
   [2/106] Building C object src\CMakeFiles\glfw.dir\input.c.obj
   [3/106] Building C object src\CMakeFiles\glfw.dir\monitor.c.obj
   [4/106] Building C object src\CMakeFiles\glfw.dir\init.c.obj
   [5/106] Building C object src\CMakeFiles\glfw.dir\vulkan.c.obj
   [6/106] Building C object src\CMakeFiles\glfw.dir\platform.c.obj
   [7/106] Building C object src\CMakeFiles\glfw.dir\window.c.obj
   [8/106] Building C object src\CMakeFiles\glfw.dir\egl_context.c.obj
   [9/106] Building C object src\CMakeFiles\glfw.dir\null_init.c.obj
   [10/106] Building C object src\CMakeFiles\glfw.dir\osmesa_context.c.obj
   [11/106] Building C object src\CMakeFiles\glfw.dir\null_joystick.c.obj
   [12/106] Building C object src\CMakeFiles\glfw.dir\null_monitor.c.obj
   [13/106] Building C object src\CMakeFiles\glfw.dir\win32_time.c.obj
   [14/106] Building C object src\CMakeFiles\glfw.dir\null_window.c.obj
   [15/106] Building C object src\CMakeFiles\glfw.dir\win32_thread.c.obj
   [16/106] Building C object src\CMakeFiles\glfw.dir\win32_module.c.obj
   [17/106] Building C object src\CMakeFiles\glfw.dir\win32_init.c.obj
   [18/106] Building C object src\CMakeFiles\glfw.dir\win32_joystick.c.obj
   [19/106] Building C object src\CMakeFiles\glfw.dir\win32_monitor.c.obj
   [20/106] Building C object src\CMakeFiles\glfw.dir\win32_window.c.obj
   [21/106] Building RC object examples\CMakeFiles\gears.dir\glfw.rc.res
   [22/106] Building C object src\CMakeFiles\glfw.dir\wgl_context.c.obj
   [23/106] Building C object examples\CMakeFiles\boing.dir\boing.c.obj
   [24/106] Linking C static library src\glfw3.lib
   [25/106] Building C object examples\CMakeFiles\gears.dir\gears.c.obj
   [26/106] Building RC object examples\CMakeFiles\boing.dir\glfw.rc.res
```
`glfw` is a correct name for the target:
```cmake
add_library(glfw ${GLFW_LIBRARY_TYPE}
                 "${GLFW_SOURCE_DIR}/include/GLFW/glfw3.h"
                 "${GLFW_SOURCE_DIR}/include/GLFW/glfw3native.h"
                 internal.h platform.h mappings.h
                 context.c init.c input.c monitor.c platform.c vulkan.c window.c
                 egl_context.c osmesa_context.c null_platform.h null_joystick.h
                 null_init.c null_monitor.c null_window.c null_joystick.c)
```
and all those tests and examples are separate targets:
```cmake
add_executable(boing WIN32 MACOSX_BUNDLE boing.c ${ICON} ${GLAD_GL})
add_executable(gears WIN32 MACOSX_BUNDLE gears.c ${ICON} ${GLAD_GL})
add_executable(heightmap WIN32 MACOSX_BUNDLE heightmap.c ${ICON} ${GLAD_GL})
add_executable(offscreen offscreen.c ${ICON} ${GLAD_GL})
add_executable(particles WIN32 MACOSX_BUNDLE particles.c ${ICON} ${TINYCTHREAD} ${GETOPT} ${GLAD_GL})
add_executable(sharing WIN32 MACOSX_BUNDLE sharing.c ${ICON} ${GLAD_GL})
add_executable(splitview WIN32 MACOSX_BUNDLE splitview.c ${ICON} ${GLAD_GL})
add_executable(triangle-opengl WIN32 MACOSX_BUNDLE triangle-opengl.c ${ICON} ${GLAD_GL})
add_executable(triangle-opengles WIN32 MACOSX_BUNDLE triangle-opengles.c ${ICON} ${GLAD_GLES2})
add_executable(wave WIN32 MACOSX_BUNDLE wave.c ${ICON} ${GLAD_GL})
add_executable(windows WIN32 MACOSX_BUNDLE windows.c ${ICON} ${GLAD_GL})
```
```cmake
add_executable(allocator allocator.c ${GLAD_GL})
add_executable(clipboard clipboard.c ${GETOPT} ${GLAD_GL})
add_executable(events events.c ${GETOPT} ${GLAD_GL})
add_executable(msaa msaa.c ${GETOPT} ${GLAD_GL})
add_executable(glfwinfo glfwinfo.c ${GETOPT} ${GLAD_GL} ${GLAD_VULKAN})
add_executable(iconify iconify.c ${GETOPT} ${GLAD_GL})
add_executable(monitors monitors.c ${GETOPT} ${GLAD_GL})
add_executable(reopen reopen.c ${GLAD_GL})
add_executable(cursor cursor.c ${GLAD_GL})

add_executable(empty WIN32 MACOSX_BUNDLE empty.c ${TINYCTHREAD} ${GLAD_GL})
add_executable(gamma WIN32 MACOSX_BUNDLE gamma.c ${GLAD_GL})
add_executable(icon WIN32 MACOSX_BUNDLE icon.c ${GLAD_GL})
add_executable(inputlag WIN32 MACOSX_BUNDLE inputlag.c ${GETOPT} ${GLAD_GL})
add_executable(joysticks WIN32 MACOSX_BUNDLE joysticks.c ${GLAD_GL})
add_executable(tearing WIN32 MACOSX_BUNDLE tearing.c ${GLAD_GL})
add_executable(threads WIN32 MACOSX_BUNDLE threads.c ${TINYCTHREAD} ${GLAD_GL})
add_executable(timeout WIN32 MACOSX_BUNDLE timeout.c ${GLAD_GL})
add_executable(title WIN32 MACOSX_BUNDLE title.c ${GLAD_GL})
add_executable(triangle-vulkan WIN32 triangle-vulkan.c ${GLAD_VULKAN})
add_executable(window WIN32 MACOSX_BUNDLE window.c ${GLAD_GL})
```

**Comment by rewrking** [comment]
Well this is embarrassing. `targets` is not read in if an array is used. For now, you can add this:
```json
"glfw": {
	"kind": "cmakeProject",
	"location": "${externalDir}/glfw",
	"targets": "glfw"
}
```

or this:

```json
"glfw": {
	"kind": "cmakeProject",
	"location": "${externalDir}/glfw",
	"defines": [
		"GLFW_BUILD_EXAMPLES=OFF",
		"GLFW_BUILD_TESTS=OFF",
		"GLFW_BUILD_DOCS=OFF"
	]
}
```

**Comment by thatbakamono** [comment]
> Well this is embarrassing. `targets` is not read in if an array is used. For now, you can add this:
> 
> ```json
> "glfw": {
> 	"kind": "cmakeProject",
> 	"location": "${externalDir}/glfw",
> 	"targets": "glfw"
> }
> ```
> 
> or this:
> 
> ```json
> "glfw": {
> 	"kind": "cmakeProject",
> 	"location": "${externalDir}/glfw",
> 	"defines": [
> 		"GLFW_BUILD_EXAMPLES=OFF",
> 		"GLFW_BUILD_TESTS=OFF",
> 		"GLFW_BUILD_DOCS=OFF"
> 	]
> }
> ```

Oh, okay, I don't need more than one target so it isn't really an issue for me, at least now.

**Comment by rewrking** [comment]
Fixed in [v0.4.2](https://github.com/chalet-org/chalet/releases/tag/v0.4.2)
